### ISSUE TITLE
Bug: `no-missing-link-fragments` false negatives when document starts with an HTML tag

### ISSUE THREAD

**Issue by fasttime** [eslint/markdown#470]
### Environment

ESLint version: v9.31.0
@eslint/markdown version: 7.0.0
Node version: 24.4.1
npm version: 11.4.2
Operating System: Windows 11

### Which language are you using?

commonmark

### What did you do?

<details>
<summary>Configuration</summary>

```
import markdown from "@eslint/markdown";

export default [{
	files: ["**/*.md"],
	language: "markdown/commonmark",
	plugins: { markdown },
	rules: { "markdown/no-missing-link-fragments": "error" },
}];
```
</details>

```md
<h1>Title</h1>
[Link](#a-link)
```

```md
<br />
[Test](#test)
```

### What did you expect to happen?

`no-missing-link-fragments` should report a problem for each snippet.

### What actually happened?

No problems are being reported.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-blxjnlht

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by fasttime** [comment]
Ah, I see! Thanks @lumirlumir for the clarification.
